yt/data_objects/construction_data_containers.py

```diff
@@ -2328,8 +2330,8 @@ def get_data(self, fields = None):
         elif isinstance(fields, list):
             fields = fields[0]
 
-        sph_ptype = getattr(self.ds, '_sph_ptype', 'None')
-        if fields[0] == sph_ptype:
+        sph_ptypes = getattr(self.ds, '_sph_ptypes', 'None')
```

Comment:
Would `getattr(self.ds, '_sph_ptypes', [])` (or something like that) be better? I don't think it's likely to run into an edge case that `fields[0] in 'None'` would be `True` unexpectedly. So just a minor suggestion.

Reply:
Here I'm not modifying the original logic of the code, so we'd have to look carefully into the  original reason for this line.

Reply:
Yeah, I think both the original and current code should work 99.99% of the time. The original logic would only be flawed if the queried field happens to be named like `('None', 'XXX')`, and then it could pass the test `fields[0] == sph_ptype` unexpectedly. But that should be an extremely rare case if not impossible. In the modified code, the same could happen. That's why I made the suggestion.